File: src/prod/src/data/utilities/PrimeLockRequest.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace Data::Utilities;

#define PRIMELOCKREQUEST_TAG 'trLP'

PrimeLockRequest::PrimeLockRequest(
    __in LockManager& lockManager,
    __in LockMode::Enum lockMode)
    : lockManagerSPtr_(&lockManager),
    lockMode_(lockMode)
{    
}

PrimeLockRequest::~PrimeLockRequest()
{
}

NTSTATUS
PrimeLockRequest::Create(
    __in LockManager& lockManager,
    __in LockMode::Enum lockMode,
    __in KAllocator& allocator,
    __out PrimeLockRequest::SPtr& result)
{
    NTSTATUS status;
    auto output = _new(PRIMELOCKREQUEST_TAG, allocator) PrimeLockRequest(lockManager, lockMode);

    if (!output)
    {
        return STATUS_INSUFFICIENT_RESOURCES;
    }

    status = output->Status();
    if (!NT_SUCCESS(status))
    {
        return status;
    }

    result = Ktl::Move(output);
    return STATUS_SUCCESS;
}

File: src/prod/src/managed/Microsoft.ServiceFabric.Data.Impl/ReplicatedStore/LockManager/PrimeLockRequest.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Store
{
    /// <summary>
    /// Contains lock manager and lock mode associated with a prime lock.
    /// </summary>
    internal struct PrimeLockRequest
    {
        private LockManager lockManager;
        private LockMode lockMode;

        /// <summary>
        ///  Initializes a new instance of the <see cref="PrimeLockRequest"/> class. 
        /// </summary>
        /// <param name="lockManager"></param>
        /// <param name="lockMode"></param>
        public PrimeLockRequest(LockManager lockManager, LockMode lockMode)
        {
            this.lockManager = lockManager;
            this.lockMode = lockMode;
        }

        /// <summary>
        /// Lock manager associated with the lock request.
        /// </summary>
        public LockManager LockManager
        {
            get { return this.lockManager; }
        }

        /// <summary>
        /// Lock mode for the lock request.
        /// </summary>
        public LockMode LockMode
        {
            get { return this.lockMode; }
        }
    }
}